authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHelper.java

```diff
@@ -56,83 +55,124 @@ public class RangerHelper {
-    this.rangerAuthorizationPlugin = rangerAuthorizationPlugin;
-    switch (catalogProvider) {
-      case "hive":
-        initPrivilegesMapping();
-        initOwnerPrivileges();
-        initPolicySearchKeys();
-        initPreciseFilterKeys();
-        break;
-      default:
-        throw new IllegalArgumentException(
-            "Authorization plugin unsupported catalog provider: " + catalogProvider);
-    }
+  private List<String> policySearchKeys = new ArrayList<>();
+
+  private List<String> policyResourceDefines = new ArrayList<>();
+
+  private final RangerClientExtend rangerClient;
+  private final String rangerAdminName;
+  private final String rangerServiceName;
+  private AuthorizationConfig authorizationConfig = null;
+
+  public RangerHelper(
+      String catalogProvider,
+      RangerClientExtend rangerClient,
+      String rangerAdminName,
+      String rangerServiceName) {
+    this.rangerClient = rangerClient;
+    this.rangerAdminName = rangerAdminName;
+    this.rangerServiceName = rangerServiceName;
+
+    this.authorizationConfig = AuthorizationConfig.loadConfig(catalogProvider);
+    initAuthorizationConfig(authorizationConfig);
   }
 
-  /** Initial mapping Gravitino privilege name to the underlying authorization system privileges. */
-  private void initPrivilegesMapping() {
-    privilegesMapping =
-        ImmutableMap.<Privilege.Name, Set<String>>builder()
-            .put(
-                Privilege.Name.CREATE_SCHEMA,
-                ImmutableSet.of(RangerDefines.ACCESS_TYPE_HIVE_CREATE))
-            .put(
-                Privilege.Name.CREATE_TABLE, ImmutableSet.of(RangerDefines.ACCESS_TYPE_HIVE_CREATE))
-            .put(
-                Privilege.Name.MODIFY_TABLE,
-                ImmutableSet.of(
-                    RangerDefines.ACCESS_TYPE_HIVE_UPDATE,
-                    RangerDefines.ACCESS_TYPE_HIVE_ALTER,
-                    RangerDefines.ACCESS_TYPE_HIVE_WRITE))
-            .put(
-                Privilege.Name.SELECT_TABLE,
-                ImmutableSet.of(
-                    RangerDefines.ACCESS_TYPE_HIVE_READ, RangerDefines.ACCESS_TYPE_HIVE_SELECT))
-            .build();
+  @VisibleForTesting
+  RangerHelper(AuthorizationConfig authorizationConfig) {
+    this.rangerClient = null;
+    this.rangerAdminName = null;
+    this.rangerServiceName = null;
+
+    this.authorizationConfig = authorizationConfig;
+    initAuthorizationConfig(authorizationConfig);
   }
 
-  /** Initial Owner privileges */
-  private void initOwnerPrivileges() {
-    ownerPrivileges = ImmutableSet.of(RangerDefines.ACCESS_TYPE_HIVE_ALL);
+  /**
+   * Initial mapping Gravitino privilege name to the underlying authorization system privileges.
+   * <br>
+   * Initial Owner privileges. <br>
+   * Initial Ranger policy search key defines. <br>
+   * Initial precise filter key defines. <br>
+   */
+  private void initAuthorizationConfig(AuthorizationConfig authorizationConfig) {
```

Comment:
Should we provide default values as before, it's too complicated to set the mapping for most users and only those users that have special requirement will configure this value.

Reply:
Good idea, I improved it, Please help me review this PR, again.